File: ch02/python/anthropic-claude-system-prompt/anthropic-claude-system-prompt.py
"""
(C) Copyright 2026 Boni Garcia (https://bonigarcia.github.io/)
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
 http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
"""
from anthropic import Anthropic


def query_model(instructions: str | None,
                prompt: str,
                model: str = "claude-sonnet-4-6",
                temperature: float = 0) -> str:
    """Send a text prompt to an Anthropic model and return the text response."""

    client = Anthropic()  # ANTHROPIC_API_KEY should be set as an environment variable

    params = {
        "model": model,
        "max_tokens": 1024,
        "temperature": temperature,
        "messages": [
            {"role": "user", "content": prompt}
        ]
    }
    if instructions:
        params["system"] = instructions

    response = client.messages.create(**params)
    return response.content[0].text


if __name__ == "__main__":
    instructions = """
    You are a strict grammar teacher.
    Always respond in one sentence and correct any mistakes.
    """
    prompt = "Explain me what is context engineering in simple words"

    print("=== With system instructions ===")
    response = query_model(instructions, prompt)
    print("User query:", prompt)
    print("Response:", response)

    print("\n=== With only user prompt ===")
    response = query_model(None, prompt)
    print("User query:", prompt)
    print("Response:", response)

File: ch02/javascript/anthropic-claude-system-prompt/anthropic-claude-system-prompt.js
/*
(C) Copyright 2026 Boni Garcia (https://bonigarcia.github.io/)
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
 http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
import Anthropic from '@anthropic-ai/sdk';

async function queryModel(instructions, userPrompt, model = "claude-sonnet-4-6", temperature = 0) {
    //  ANTHROPIC_API_KEY should be set as an environment variable
    const client = new Anthropic();

    const params = {
        model: model,
        max_tokens: 1024,
        temperature: temperature,
        messages: [
            { role: "user", content: userPrompt }
        ]
    };
    if (instructions) {
        params.system = instructions;
    }

    const message = await client.messages.create(params);

    return message.content[0].text;
}

const instructions = "You are a strict grammar teacher. Always respond in one sentence and correct any mistakes.";
const prompt = "Explain me what is context engineering in simple words";

console.log("=== With system prompt ===");
queryModel(instructions, prompt)
    .then(response => {
        console.log("User:", prompt);
        console.log("AI:", response);

        console.log("\n=== With only user prompt ===");
        return queryModel(null, prompt);
    })
    .then(response => {
        console.log("User:", prompt);
        console.log("AI:", response);
    })
    .catch(error => {
        console.error("Error:", error);
    });
